### PR TITLE
feat(ui5-menu-item): add endContent slot

### PR DESCRIPTION
**Pull request by NHristov-sap**
New slot `endContent` is introduced in `ui5-menu-item` component. This slot accepts components that are displayed at the end of the Menu Item if it does not have sub-menu. If there are components slotted and also additionalText is being set, the additionalText will not be shown as well. 

It is strongly recommended  to slot only icon-only `ui5-button`, `ui5-icon` or `ui5-link` components.

BGSOFUIBALKAN-8160
#6350

### CODE REVIEW

File: packages/main/src/ListItemBase.ts
Comment (kgogov) [inline packages/main/src/ListItemBase.ts:1]: From the list item's side, the changes look fine to me.

File: packages/main/src/MenuItem.ts [hunk 1]
```diff
@@ -158,6 +158,16 @@ class MenuItem extends ListItem {
 	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
 	items!: Array<MenuItem>;
 
+	/**
+	 * Defines the components that should be displayed at the end of the menu item.
+	 *
+	 * **Note:** It is highly recommended to slot only components of type `ui5-button` (icon-only),
```
Comment (unazko) [inline packages/main/src/MenuItem.ts:164]: Not quite sure about this note. Maybe we should only keep to what is recommended and what is the intention of providing this slot.

Comment (NHristov-sap) [reply packages/main/src/MenuItem.ts:164]: Done

**Review by unazko — APPROVED** [review]

File: packages/main/src/MenuItem.ts [hunk 2]
```diff
@@ -158,6 +158,15 @@ class MenuItem extends ListItem {
 	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
 	items!: Array<MenuItem>;
 
+	/**
+	 * Defines the components that should be displayed at the end of the menu item.
+	 *
+	 * **Note:** It is highly recommended to slot only components of type `ui5-button`,`ui5-link` or `ui5-icon`.
+	 * @public
+	 */
+	@slot({ type: HTMLElement, invalidateOnChildChange: true })
```
Comment (vladitasev) [inline packages/main/src/MenuItem.ts:167]: `invalidateOnChildChange` is redundant. Why did you set it? Every slot is already invalidated if you add new/remove existing children. This setting is only necessary if you want behavior that depends on children changing their properties, f.e. the list must invalidate itself if a list item changes its `mode` property, and is usually necessary for components that render abstract elements. Here I don't see any logic that depends on the state of the children inside the new slot.

File: packages/main/src/MenuItem.ts [hunk 3]
```diff
@@ -158,6 +158,15 @@ class MenuItem extends ListItem {
 	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
 	items!: Array<MenuItem>;
 
+	/**
+	 * Defines the components that should be displayed at the end of the menu item.
+	 *
+	 * **Note:** It is highly recommended to slot only components of type `ui5-button`,`ui5-link` or `ui5-icon`.
```
Comment (vladitasev) [inline packages/main/src/MenuItem.ts:164]: please add the standard wording we have for these slots "in order to preserve the intended design" at the end of the sentence.

**Review by vladitasev — CHANGES_REQUESTED** [review]
This API looks a bit strange: you have 2 slots, but they are mutually exclusive, and also an `additionalText` property that is exclusive with the slots.
 1) If you have passed submenu items, the "arrow" icon will be rendered
 2) if you have passed end content, it will be displayed
 3) lowest prio - the `additionalText` property will be displayed (this one is documented properly explaining that it's exclusive with the submenu)
 
 There are 2 solutions:
  - document the slots as mutually exclusive
  - use the same "default" slot for both purposes: if you pass menu items, then work as before. But if you pass other content, act as the current "endContent". But maybe this is not ideal because what would happen if people pass a mix of submenus and arbitrary content? We'd still have to document how it would behave in that scenario. So maybe the current implementation is better

File: packages/main/src/MenuItem.ts (hunk 2 above)
Comment (NHristov-sap) [reply packages/main/src/MenuItem.ts:167]: Ok, will remove it.

File: packages/main/src/MenuItem.ts (hunk 3 above)
Comment (NHristov-sap) [reply packages/main/src/MenuItem.ts:164]: Done.

File: packages/main/src/MenuItem.hbs [hunk 4]
```diff
@@ -20,6 +20,8 @@
 			>
 			</ui5-icon>
 		</div>
+	{{else if hasEndContent}}
```
[the remark this answers is not in the document]
Comment (NHristov-sap) [reply packages/main/src/MenuItem.hbs:23]: Ok, will update the documentation. One slot for all is not acceptable, as there are plans to add some semantics on additionalText that are not applicable to endContent items.

**Comment by NHristov-sap** [comment]
> There are 2 solutions:
> 
> * document the slots as mutually exclusive
> * use the same "default" slot for both purposes: if you pass menu items, then work as before. But if you pass other content, act as the current "endContent". But maybe this is not ideal because what would happen if people pass a mix of submenus and arbitrary content? We'd still have to document how it would behave in that scenario. So maybe the current implementation is better

Using one slot for all three purposes is not an option here, so I have updated the documentation to explain the priority of what is displayed at the end of the menu item.

**Review by tsanislavgatev — APPROVED** [review]
Looks good to me, but let's add UA to review the PR. We can merge it and address the coments later, but we can use this PR for the review

**Pull request merged**